test/functional/managedclusterinfo_test.go

```diff
@@ -214,21 +232,79 @@ var _ = Describe("Metrics", func() {
 					return fmt.Errorf("Unexpect queryResult %v", queryResult)
 				}
 
+				return nil
+			}).Should(BeNil())
+		})
+		By("Query metrics by sum(acm_managed_cluster_labels) by (cloud)", func() {
+			Eventually(func() error {
+				resp, b, err := queryMetrics("sum(acm_managed_cluster_labels)+by+(cloud)")
+				klog.Infof("Get Empty Metrics response: %v", b)
+				if err != nil || resp.StatusCode != http.StatusOK {
+					return fmt.Errorf("Failed to query metrics %v: %v", resp.StatusCode, err)
+				}
+
+				queryResult := QueryResult{}
+				if err := json.Unmarshal([]byte(b), &queryResult); err != nil {
+					return err
+				}
+
+				expectResult := map[string]string{"Amazon": "3", "": "1"}
+				actualResult := map[string]string{}
+				for _, r := range queryResult.Data.Result {
+					key := r.Metric.Cloud + r.Metric.CreatedVia + r.Metric.Vendor
+					actualResult[key] = r.Value[1].(string)
+				}
+
+				klog.Infof("expect result %v", expectResult)
+				klog.Infof("actual result %v", actualResult)
+				if !reflect.DeepEqual(expectResult, actualResult) {
+					return fmt.Errorf("Unexpect queryResult %v", queryResult)
+				}
+
+				return nil
+			}).Should(BeNil())
+		})
+		By("Query metrics by sum(acm_managed_cluster_labels) by (hub_cluster_id)", func() {
+			Eventually(func() error {
+				resp, b, err := queryMetrics("sum(acm_managed_cluster_labels)+by+(hub_cluster_id)")
+				klog.Infof("Get Empty Metrics response: %v", b)
+				if err != nil || resp.StatusCode != http.StatusOK {
+					return fmt.Errorf("Failed to query metrics %v: %v", resp.StatusCode, err)
+				}
+
+				queryResult := QueryResult{}
+				if err := json.Unmarshal([]byte(b), &queryResult); err != nil {
+					return err
+				}
+
+				expectResult := map[string]string{"": "4"}
+				actualResult := map[string]string{}
+				for _, r := range queryResult.Data.Result {
+					key := r.Metric.Cloud + r.Metric.CreatedVia + r.Metric.Vendor
+					actualResult[key] = r.Value[1].(string)
+				}
+
+				klog.Infof("expect result %v", expectResult)
+				klog.Infof("actual result %v", actualResult)
+				if !reflect.DeepEqual(expectResult, actualResult) {
+					return fmt.Errorf("Unexpect queryResult %v", queryResult)
+				}
+
 				return nil
 			}).Should(BeNil())
 		})
 	})
 
 	It("Get Empty Metrics", func() {
 		By("Getting metrics", func() {
-			Eventually(func() string {
+			Eventually(func() int {
```

Comment:
I think we still need to compare the string content, any reason to change to int?

Reply:
@haoqing0110 I changed it to `int` because when the response string is being compared, it detects the slightest change for the `acm_managed_cluster_label` name metric:

```bash
# Expected
acm_managed_cluster_labels{hub_cluster_id="787e5a35-c911-4341-a2e7-65c415147aeb",managed_cluster_id="local_cluster_id",cloud="Amazon",openshiftVersion="4.3.1",vendor="OpenShift"} 1

# Received
acm_managed_cluster_labels{hub_cluster_id="787e5a35-c911-4341-a2e7-65c415147aeb",managed_cluster_id="local_cluster_id",vendor="OpenShift",cloud="Amazon",openshiftVersion="4.3.1"} 1
```

Even though the string contains the same content, if it's out of order, this will cause the test to fail because the strings aren't considered equal. If we used `string`, then we would need to add more complex logic to the test. If that is the route that is recommended, I would suggest focusing on refactoring the test case logic within a separate PR.

Reply:
@dislbenn I see. At least for the metrics `acm_managed_cluster_info` still need to test the returned string, this test is to avoid some breaking changes, checking the line number is not enough.